src/main/java/org/breedinginsight/daos/ProgramUploadDAO.java

```diff
@@ -0,0 +1,112 @@
+/*
+ * See the NOTICE file distributed with this work for additional information
+ * regarding copyright ownership.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.breedinginsight.daos;
+
+import org.breedinginsight.dao.db.enums.UploadType;
+import org.breedinginsight.dao.db.tables.BiUserTable;
+import org.breedinginsight.dao.db.tables.daos.BatchUploadDao;
+import org.breedinginsight.model.*;
+import org.breedinginsight.model.User;
+import org.jooq.*;
+
+import javax.inject.Inject;
+import javax.inject.Singleton;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.UUID;
+
+import static org.breedinginsight.dao.db.Tables.*;
+
+@Singleton
+public class ProgramUploadDAO extends BatchUploadDao {
+
+    private DSLContext dsl;
+
+    @Inject
+    public ProgramUploadDAO(Configuration config, DSLContext dsl) {
+        super(config);
+        this.dsl = dsl;
+    }
+
+    public List<ProgramUpload> getUploads(UUID programId, UUID userId, UploadType type) {
+
+        List<Record> records = getUploadsQuery()
+                .where(BATCH_UPLOAD.PROGRAM_ID.eq(programId)
+                        .and(BATCH_UPLOAD.USER_ID.eq(userId))
+                        .and(BATCH_UPLOAD.TYPE.eq(type)))
+                .fetch();
+
+        return parseRecords(records);
+
+    }
+
+    public Optional<ProgramUpload> getUploadById(UUID id) {
+        List<Record> records = getUploadsQuery()
+                .where(BATCH_UPLOAD.ID.eq(id))
+                .fetch();
+
+        if (records.isEmpty()) {
+            return Optional.empty();
+        } else {
+            return Optional.of(parseRecords(records).get(0));
+        }
+
+    }
+
+    private SelectOnConditionStep<Record> getUploadsQuery() {
+
+        BiUserTable createdByUser = BI_USER.as("createdByUser");
+        BiUserTable updatedByUser = BI_USER.as("updatedByUser");
+
+        return dsl.select()
+                .from(BATCH_UPLOAD)
+                .leftJoin(PROGRAM).on(BATCH_UPLOAD.PROGRAM_ID.eq(PROGRAM.ID))
+                .leftJoin(BI_USER).on(BATCH_UPLOAD.USER_ID.eq(BI_USER.ID))
+                .leftJoin(createdByUser).on(BATCH_UPLOAD.CREATED_BY.eq(createdByUser.ID))
+                .leftJoin(updatedByUser).on(BATCH_UPLOAD.UPDATED_BY.eq(updatedByUser.ID));
```

Comment:
Ya, I think this came up in a previous review as well. I think we have some queries that I copied from that hadn't been updated because they were before that feedback. I'll update these.

Reply:
Pushed changes.